site/src/modules/provisioners/ProvisionerGroup.tsx

```diff
@@ -0,0 +1,213 @@
+import { useTheme } from "@emotion/react";
+import Business from "@mui/icons-material/Business";
+import Person from "@mui/icons-material/Person";
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import type { BuildInfoResponse, ProvisionerDaemon } from "api/typesGenerated";
+import { DropdownArrow } from "components/DropdownArrow/DropdownArrow";
+import { Pill } from "components/Pill/Pill";
+import { type FC, useState } from "react";
+import { createDayString } from "utils/createDayString";
+import { ProvisionerTag } from "./ProvisionerTag";
+
+type ProvisionerGroupType = "builtin" | "psk" | "key";
+
+interface ProvisionerGroupProps {
+	readonly buildInfo?: BuildInfoResponse;
+	readonly keyName?: string;
+	readonly type: ProvisionerGroupType;
+	readonly provisioners: ProvisionerDaemon[];
+}
+
+export const ProvisionerGroup: FC<ProvisionerGroupProps> = ({
+	buildInfo,
+	keyName,
+	type,
+	provisioners,
+}) => {
+	const [provisioner] = provisioners;
```

Comment:
Should there be a check to make sure that the `provisioners` array isn't empty, or is that basically guaranteed not to happen?

If there are multiple provisioners, is there anything implied by the order they appear in? As in, is the first provisioner always the "main" one?

Reply:
currently, it should never be passed an empty `provisioners` array, but I'm gonna refactor this to make it nicer.